libsolidity/ast/Types.cpp

```diff
@@ -404,10 +404,19 @@ Result<FunctionDefinition const*> Type::userDefinedOperator(Token _token, ASTNod
 				function.libraryFunction() ? function.typeViaContractName() : function.type()
 			);
 			solAssert(functionType && !functionType->parameterTypes().empty());
-			solAssert(isImplicitlyConvertibleTo(*functionType->parameterTypes().front()));
+
+			Type const* expectedReturnType =
```

Comment:
I'm not sure we should do such complicated things here - I would have a better feeling if we just did syntactic checks here.

Reply:
I have updated the code so a return parameter type checking is now part of the TypeChecker.